dockstore-client/src/main/java/io/dockstore/client/cli/nested/NotificationsClients/NotificationsClient.java

```diff
@@ -0,0 +1,120 @@
+package io.dockstore.client.cli.nested.NotificationsClients;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import com.google.gson.Gson;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author gluu
+ * @since 12/01/18
+ */
+public class NotificationsClient {
+    public static final String PROVISION_INPUT = "provision-in";
+    public static final String RUN = "workflow-start";
+    public static final String PROVISION_OUTPUT = "provision-out";
+    public static final String COMPLETED = "workflow-complete";
+    protected static final String USERNAME = "Dockstore CLI";
+    private static final Logger LOG = LoggerFactory.getLogger(NotificationsClient.class);
+    protected String hookURL;
+    protected String uuid;
+    protected boolean disabled = false;
+
+
+    public NotificationsClient(String hookURL, String uuid) {
+        this.hookURL = hookURL;
+        this.uuid = uuid;
+        boolean invalidHookURL = (hookURL == null || hookURL.isEmpty());
+        boolean invalidUUID = (uuid == null || uuid.isEmpty());
+        if (invalidHookURL || invalidUUID) {
+            disabled = true;
```

Comment:
Sorry, forgot to mention last time. 
If a uuid is not provided, just generate one.